vtrace/platforms/win32.py

```diff
@@ -2210,3 +2223,116 @@ def getCacheSyms(self):
             ret.append( (addr-self.loadbase, size, name, stype) )
         return ret
 
+    def genSymbols(self, symopts=None, symtypes=None):
+        '''
+        Generator of symbols... does not cache symbols.  This is good for PDBs of unusual size (POUSes).
+        '''
+        oldsymopts = self.symopts
+        if symopts != None:
+            self.symopts = symopts
+        
+        self.kickOffSymEnumSymbols()
+        self._symEnumLock.acquire()
+
+        while self.symbol != -1:
+            #print 'genSymbols: _symEnumLock.acquire()'
+            yield self.symbol
+            #print 'genSymbols: _symEnumHandling.release()'
+            self._symEnumHandling.release()
+            self._symEnumLock.acquire()
+
+        self.symopts = oldsymopts
+
+    def genTypes(self, symopts=None, symtypes=None):
+        '''
+        Generator of Types
+        '''
+        oldsymopts = self.symopts
+        if symopts != None:
+            self.symopts = symopts
+    
+        self.kickOffSymEnumTypes()
+
+        self._typeEnumLock.acquire()
```

Comment:
this lock is never released on the eventual generator breakout from the loop? ( ie, genTypes would never be able to be called again )